src/components/admin/lib/__tests__/article-validators.test.js

```diff
@@ -0,0 +1,87 @@
+import { hasNonHttpsURL, returnsFirstRelativeURL } from '../article-validators';
+
+describe('hasNonHttpsURL', () => {
+  it('Returns true if it finds an http (not s) link in an href', () => {
+    expect(
+      hasNonHttpsURL('<a href=\'http://www.w3schools.com\'> Title\'hello\'</a>')
+    ).toBe(true);
+    expect(
+      hasNonHttpsURL(
+        '<a href=\'http://www.w3schools.com/\' target=\'_blank\'>Visit W3Schools!</a>'
+      )
+    ).toBe(true);
+  });
+  it('Returns false if it finds an https in an href', () => {
+    expect(
+      hasNonHttpsURL('<a href=\'https://www.w3schools.com\'> Title\'hello\'</a>')
+    ).toBe(false);
+  });
+  it('Returns true if it finds an http (not s) link in an img tag', () => {
+    expect(
+      hasNonHttpsURL('<img src=\'http://image.com\' alt=\'Mountain View\'>')
+    ).toBe(true);
+    expect(
+      hasNonHttpsURL(
+        '<img src=\'http://image.com\' alt=\'HTML tutorial\' style=\'width:42px;height:42px;border:0;\'>'
+      )
+    ).toBe(true);
+  });
+  it('Returns false if it finds an https in an img tag', () => {
+    expect(
+      hasNonHttpsURL('<img src=\'https://image.com\' alt=\'Mountain View\'>')
+    ).toBe(false);
+  });
+  it('Returns false when encountering a url in the text that is not a link', () => {
+    expect(hasNonHttpsURL('<p>http://www.google.com</p>')).toBe(false);
+    expect(
+      hasNonHttpsURL(
+        '<a href=\'https://www.w3schools.com\'> Title\'http://www.google.com\'</a>'
+      )
+    ).toBe(false);
+  });
+});
+
+describe('returnsFirstRelativeURL', () => {
+  it('Returns null if an href has an absolute URL', () => {
```

Comment:
The wording here and below of `if it has an absolute URL` is a bit misleading, as it actually has more to do with whether there are relative urls, right? So maybe more something along the lines of either `doesn't match absolute URLs` or `Returns null if there are no relative URLs`, where I think (where applicable` that the first one is even clearer but both work